test/e2e/framework/globalegressip.go

```diff
@@ -0,0 +1,116 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	submarinerClientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var globalEgressIPGVR = &schema.GroupVersionResource{
+	Group:    "submariner.io",
+	Version:  "v1",
+	Resource: "globalegressips",
+}
+
+func globalEgressIPClient(cluster framework.ClusterIndex, namespace string) dynamic.ResourceInterface {
+	return framework.DynClients[cluster].Resource(*globalEgressIPGVR).Namespace(namespace)
+}
+
+func CreateGlobalEgressIP(cluster framework.ClusterIndex, name, namespace string,
+	selector *metav1.LabelSelector) (*submarinerv1.GlobalEgressIP, error) {
+	egressIPSpec := &submarinerv1.GlobalEgressIP{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	if selector != nil {
+		egressIPSpec.Spec = submarinerv1.GlobalEgressIPSpec{
+			PodSelector: selector,
+		}
+	}
+
+	config := framework.RestConfigs[cluster]
+	client, err := submarinerClientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return framework.AwaitUntil("create GlobalEgressIP", func() (interface{}, error) {
+		egressIP, err := client.SubmarinerV1().GlobalEgressIPs(namespace).Create(context.TODO(), egressIPSpec, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			err = client.SubmarinerV1().GlobalEgressIPs(namespace).Delete(context.TODO(), egressIPSpec.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return nil, err
+			}
+
+			egressIP, err = client.SubmarinerV1().GlobalEgressIPs(namespace).Create(context.TODO(), egressIPSpec, metav1.CreateOptions{})
+		}
+
+		return egressIP, err
+	}, framework.NoopCheckResult).(*submarinerv1.GlobalEgressIP), nil
+}
+
+func AwaitGlobalEgressIPs(cluster framework.ClusterIndex, name, namespace string) []string {
+	if framework.TestContext.GlobalnetEnabled {
```

Comment:
Should this even be called if globalnet isn't enabled? Also `CreateGlobalEgressIP` doesn't check the flag. I think this c heck should be left up to the caller.